threeML/test/test_fits_file.py

```diff
@@ -0,0 +1,68 @@
+
+from threeML.io.fits_file import FITSExtension, FITSFile
+import numpy as np
+import astropy.io.fits as fits
+
+import pytest
+
+
+class DUMMYEXT(FITSExtension):
+
+    def __init__(self, test_value):
+
+        data_list = [('TEST_VALUE', test_value)]
+
+        super(DUMMYEXT, self).__init__(tuple(data_list), (('EXTNAME', 'TEST', 'Extension name'),) )
+
+
+class DUMMYFITS(FITSFile):
+
+    def __init__(self, test_value):
+
+        dummy_extension = DUMMYEXT(test_value)
+
+        super(DUMMYFITS, self).__init__(fits_extensions=[dummy_extension])
+
+
+def test_fits_file():
+
+    dtypes = [np.int16,np.int32,np.int64,np.uint16,np.uint32,np.float32,np.float64]
+    dtype_keys = ['I','J','K','I','J','E','D']
+
+
+    for i, dt in enumerate(dtypes):
+
+        test_values = np.ones(10,dtype=dt)
+
+        dummy_fits = DUMMYFITS(test_value=test_values)
+
+        assert len(dummy_fits._hdu_list) == 2
+
+        assert dummy_fits.index_of('TEST') == 1
+
+        assert dummy_fits['TEST'].header['TFORM1'] == dtype_keys[i]
+
+        assert len(dummy_fits['TEST'].data['TEST_VALUE'] == test_values)
```

Comment:
doh. I always forget this. You are correct. Actually, I even had a small bug here with the len(...) argument. I'll fix it up.